kuwala/core/canvas/src/pages/DataSourcePreview.js

```diff
@@ -1,15 +1,44 @@
-import React, {useEffect, useState} from "react";
+import React, {useEffect, useMemo, useState} from "react";
 import Header from "../components/Header";
 import {useLocation, Link} from "react-router-dom";
 import {useStoreState} from "easy-peasy";
+import ReactTable from 'react-table-6';
+import "react-table-6/react-table.css";
+import "./styles/data-source-preview-table.style.css";
 
 import ListSVG from "../icons/list.svg"
 import ArrowRight from "../icons/arrow-right-solid.svg"
 import ArrowDown from "../icons/arrow-down-solid.svg"
 import FolderSVG from "../icons/folder-solid.svg"
 import TableSVG from "../icons/table-solid.svg"
 import {getSchema, getTablePreview} from "../api/DataSourceApi";
-import {data} from "autoprefixer";
+
+const Table = ({columns, data}) => {
+    const memoizedCols = useMemo(()=> {
+        return columns
+    },[]);
+
+    const memoizedRows = useMemo(()=> {
+        return data
+    },[]);
+
+    return (
+        <ReactTable
+            data={memoizedRows}
+            columns={memoizedCols}
+            defaultPageSize={20}
```

Comment:
Default page size should be 300 (see non-functional requirements in #81 and #80).